streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/IQv2EndpointToPartitionsIntegrationTest.java

```diff
@@ -0,0 +1,229 @@
+            streamOneProperties.put(StreamsConfig.GROUP_PROTOCOL_CONFIG, groupProtocolConfig);
+            if (usingStandbyReplicas) {
+                streamOneProperties.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, numStandbyReplicas);
+            }
+            streamsApplicationProperties = props(streamOneProperties);
+
+            final Properties streamTwoProperties = new Properties();
+            streamTwoProperties.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory(appId).getPath() + "-ks2");
+            streamTwoProperties.put(StreamsConfig.CLIENT_ID_CONFIG, appId + "-ks2");
+            streamTwoProperties.put(StreamsConfig.APPLICATION_SERVER_CONFIG, "localhost:3030");
+            streamTwoProperties.put(StreamsConfig.GROUP_PROTOCOL_CONFIG, groupProtocolConfig);
+            if (usingStandbyReplicas) {
+                streamTwoProperties.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, numStandbyReplicas);
+            }
+            streamsSecondApplicationProperties = props(streamTwoProperties);
+
+            final Topology topology = complexTopology();
+            try (final KafkaStreams streamsOne = new KafkaStreams(topology, streamsApplicationProperties)) {
+                IntegrationTestUtils.startApplicationAndWaitUntilRunning(streamsOne);
+                final List<StreamsMetadata> streamsMetadataAllClients = new ArrayList<>(streamsOne.metadataForAllStreamsClients());
+                assertEquals(1, streamsMetadataAllClients.size());
+                final StreamsMetadata streamsOneMetadataOne = streamsMetadataAllClients.get(0);
+                final Set<TopicPartition> topicPartitions = streamsOneMetadataOne.topicPartitions();
+                assertEquals(2020, streamsOneMetadataOne.hostInfo().port());
+                assertEquals(4, topicPartitions.size());
+                assertEquals(0, streamsOneMetadataOne.standbyTopicPartitions().size());
+
+                final long repartitionTopicTaskCount = topicPartitions.stream().filter(tp -> tp.topic().contains("-repartition")).count();
+                final long sourceTopicTaskCount = topicPartitions.stream().filter(tp -> tp.topic().contains("-input-two")).count();
+                assertEquals(2, repartitionTopicTaskCount);
+                assertEquals(2, sourceTopicTaskCount);
+                final int expectedStandbyTopicPartitionCount = usingStandbyReplicas ? 1 : 0;
+
+                try (final KafkaStreams streamsTwo = new KafkaStreams(topology, streamsSecondApplicationProperties)) {
+                    streamsTwo.start();
+                    waitForCondition(() -> KafkaStreams.State.RUNNING == streamsTwo.state() && KafkaStreams.State.RUNNING == streamsOne.state(),
+                            IntegrationTestUtils.DEFAULT_TIMEOUT,
+                            () -> "Kafka Streams one or two never transitioned to a RUNNING state.");
+
+                    waitForCondition(() ->  {
```

Comment:
Although we can infer the task assignments from the metadata, I wanted to have an explicit check here for the expected task movement.